UnityProjectDP/Assets/Scripts/Visualization/UI/ToolManager.cs

```diff
@@ -66,10 +63,56 @@ public void SelectColor(string colorID)
             SelectedColor = c;
         }
 
-        protected override void OnAwake()
+        public void SetActive(bool active)
+        {
+            if (active)
+                SelectedTool = _lastSelectedTool;
+            else
+            {
+                _lastSelectedTool = SelectedTool;
+                SelectedTool = Tool.NoToolSelected;
+            }
+        }
+
+        private void OnButtonClicked(Button clickedButton)
         {
-            base.OnAwake();
-            SelectedTool = "CameraMovement";
+            var buttonIndex = System.Array.IndexOf(buttons, clickedButton);
+
+            if (buttonIndex == -1)
+                return;
+
+            buttons.ForEach(button => button.interactable = true);
+
+            clickedButton.interactable = false;
+        }
+
+        private void SelectTool(Tool toolName)
```

Comment:
```suggestion
        private void SelectTool(Tool tool)
```
nie je to meno, ale tool (enum)